src/store/filestore.rs

```diff
@@ -280,7 +280,8 @@ impl<S: Store> ClientFilestore<S> {
                             real_dir: dir.clone(),
                             last: i,
                             location,
-                            user_attribute,
+                            // TODO: check if we can avoid that clone
```

Comment:
The `Dyn` impl could actually take an `FnOnce`, turn it into an `FnMut` that panics if invoked more than once.

Reply:
I think we should also be able to just move that part of the code out of the closure.

Reply:
Yeah, that works and also gets rid of the dir clone.